### ISSUE TITLE
Trigger "changeId" need to be generate only if the content id change

### ISSUE THREAD

**Issue by Noodlex** [jashkenas/backbone#4289]
### Affected component

The set function of backbone model

### Expected behavior

Trigger "changeId" if the changeId is different of previous

### Actual behavior

Always trigger "changeId"

### Software stack

- Backbone 1.6.0

### Steps to reproduce

1. fetch a backbone model
2. re-fetch the same backbone model with the same id
3. trigger "changeId" generate but there is no change id, it the same id

**Comment by jgonggrijp** [comment]
Thank you for reporting this. I was able to reproduce this with the following snippet of code:

```js
var attrs = {id: 1, fruit: 'banana'};
var m = new Backbone.Model(attrs);
m.on('all', console.log);
m.set(attrs);
// Logs "changeId" to the console, plus event payload.
```

Before even trying that, I already took a peek at the code and immediately got a suspicion what might be causing this. The event is triggered here:

https://github.com/jashkenas/backbone/blob/4e642088c4810938a0510fa89456ef2619f446da/backbone.js#L523-L527

The condition of the `if` goes `this.idAttribute in attrs`, which checks whether the idAttribute is specified in the attributes passed to `set`. However, that does not take into account whether the specified value is actually different from the existing `id`. A more appropriate check would be `_.contains(changes, this.idAttribute)`.

One other thing I noticed while looking at the code: `changeId` currently triggers even if the option `silent: true` is passed. Line 526 (from the snippet above) should probably be made conditional on `!silent`.
